File: go/fiber/internal/modules/account/dto/create_account_dto.go
package dto

import (
	"tcc/internal/modules/account/enum"

	"github.com/google/uuid"
)



type CreateAccountDto struct {
	Tipo      enum.AccountTypeEnum `validate:"required,oneof=CORRENTE POUPANCA"`
	IdAgencia *uuid.UUID      `validate:"omitempty,uuid"`
	IdCliente *uuid.UUID      `validate:"omitempty,uuid"`
}

File: rust/actix/src/modules/account/dto/create_account_dto.rs
use serde::{Deserialize, Serialize};
use uuid::Uuid;
use validator::Validate;
use crate::modules::account::enums::AccountTypeEnum;

#[derive(Debug, Serialize, Deserialize, Validate)]
#[serde(rename_all = "camelCase")]
pub struct CreateAccountDto {
    #[validate(required)]
    pub tipo: Option<AccountTypeEnum>,
    
    pub id_agencia: Option<Uuid>,
    
    pub id_cliente: Option<Uuid>,
}